core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -44,6 +44,9 @@ private TableProperties() {}
   public static final String MANIFEST_MERGE_ENABLED = "commit.manifest-merge.enabled";
   public static final boolean MANIFEST_MERGE_ENABLED_DEFAULT = true;
 
+  public static final String SNAPSHOT_ID_INHERITANCE_ENABLED = "commit.snapshot-id-inheritance.enabled";
```

Comment:
Minor: I would rather have a different prefix for feature flags. This does affect commit behavior, but this property doesn't signal that it breaks forward-compatibility to turn it on.

What about using `compatibility` as the initial prefix? Then we would have `compatibility.snapshot-id-inheritance.enabled`. I think that signals clearly that it affects compatibility and you should look it up in documentation.

Reply:
Done